vpr/src/route/check_rr_graph.cpp

```diff
@@ -124,16 +124,16 @@ void check_rr_graph(const t_graph_type graph_type,
             t_rr_type to_rr_type = device_ctx.rr_nodes[to_node].type();
 
             /* Only expect the following cases to have multiple edges
-             * - chan <-> chan connections 
-             * - IPIN <-> chan connections (unique rr_node for IPIN nodes on multiple sides)
-             * - OPIN <-> chan connections (unique rr_node for OPIN nodes on multiple sides)
+             * - CHAN <-> CHAN connections
+             * - CHAN  -> IPIN connections (unique rr_node for IPIN nodes on multiple sides)
+             * - OPIN  -> CHAN connections (unique rr_node for OPIN nodes on multiple sides)
              */
-            if (((to_rr_type != CHANX && to_rr_type != CHANY && rr_type != IPIN)
-                 || (rr_type != CHANX && rr_type != CHANY))
-                && ((to_rr_type != CHANX && to_rr_type != CHANY)
-                    || (rr_type != CHANX && rr_type != CHANY && rr_type != OPIN))) {
+            bool is_chan_to_chan = (rr_type == CHANX && to_rr_type == CHANY) || (rr_type == CHANY && to_rr_type == CHANX);
```

Comment:
`is_chan_to_chan` should be `(rr_type == CHANX || rr_type == CHANY) && (to_rr_type == CHANX || to_rr_type == CHANY)`.  It is legal to have CHANX <-> CHANX for adjacent tracks.

Reply:
Oh right, thanks. Fixed